### PR TITLE
[react-tracking] Update types for 8.1

### LINKED ISSUE #52105: [@types/react-tracking] outdated type for useTracking hook
# Problem
Latest version of `@types/react-tracking` have wrong type definitions for the hook `useTracking` since the library had an update and now this hooks returns a `Tracking` component that is missing in the type definitions and also accepts an object in this format `{ page: 'FooPage', action: 'childClick'}` as a parameter

- Authors: @damassi @alloy @ForbesLindesay @ferdaber @drbalm @sad-squid @chenasraf

### PR DESCRIPTION
**Pull request by chenasraf**
Hi, 
I have made updates to support the latest version of [@nytimes/react-tracking](https://github.com/nytimes/react-tracking).

Fixes #52105

The following is a breakdown of the major changes:

- add `TrackingHook` interface
- add `TrackingComponent` type
- add parameters to `useTracking`
- add `DecoratorOptions` to `Track`

Here are additional changes made:

- ~~`tslint.json` was updated because of an error when testing. I can revert this change if it is unnecessary and is only a problem on my machine. This was the error:~~
    ```
    yarn test react-tracking
    yarn run v1.22.10
    $ dtslint types react-tracking
    Error: <path>/DefinitelyTyped/node_modules/csstype/index.d.ts:1:1
    ERROR: 1:1  no-outside-dependencies  File <path>/Dev/DefinitelyTyped/node_modules/csstype/index.d.ts comes from a `node_modules` but is not declared in this type's `package.json`.  See: 
    https://github.com/Microsoft/dtslint/blob/master/docs/no-outside-dependencies.md

        at <path>/DefinitelyTyped/node_modules/dtslint/bin/index.js:207:19
        at Generator.next (<anonymous>)
        at fulfilled (<path>/DefinitelyTyped/node_modules/dtslint/bin/index.js:6:58) error Command failed with exit code 1.
    ```
- `prettier` formatted some of the lines, mostly replacing single with double quotes

PR checklist:

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nytimes/react-tracking/releases/tag/v8.1.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by EduardoAraujoB — APPROVED** [review]
LGTM

**Review by damassi — APPROVED** [review]

**Comment by damassi** [comment]
👍

File: types/react-tracking/tslint.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
     "extends": "dtslint/dt.json",
     "rules": {
         "ban-types": false,
-        "no-unnecessary-generics": false
+        "no-unnecessary-generics": false,
+        "no-outside-dependencies": false
```
Comment (weswigham) [inline types/react-tracking/tslint.json:6]: Don't disable this lint rule - instead make a PR to https://github.com/microsoft/DefinitelyTyped-tools/tree/master/packages/definitions-parser/allowedPackageJsonDependencies.txt

**Review by weswigham — CHANGES_REQUESTED** [review]

File: types/react-tracking/tslint.json (hunk 1 above)
Comment (weswigham) [reply types/react-tracking/tslint.json:6]: Although I don't see any new imports, so I don't see why it was added at all?

Comment (chenasraf) [reply types/react-tracking/tslint.json:6]: Maybe it only fails on my machine, I can make a commit to revert it and test if you think it might work.

Comment (chenasraf) [reply types/react-tracking/tslint.json]: Just pushed the revert, will resolve this conversation if the CI passes.

**Pull request merged**